src/index.ts

```diff
@@ -18,18 +18,36 @@ type Value<A> = A extends Sum.Member<any, infer B> ? B : never
 // We're going to be defining a series of very similar-looking mapped types
 // without abstraction owing to TypeScript's lack of support for higher-kinded
 // types i.e. passing generics as type arguments.
-
-type Eqs<A extends Sum.AnyMember> = {
-  readonly [B in A as Value<B> extends null ? never : Tag<B>]: Eq<Value<B>>
-}
-
-type Ords<A extends Sum.AnyMember> = {
-  readonly [B in A as Value<B> extends null ? never : Tag<B>]: Ord<Value<B>>
-}
-
-type Shows<A extends Sum.AnyMember> = {
-  readonly [B in A as Value<B> extends null ? never : Tag<B>]: Show<Value<B>>
-}
+//
+// If a sum is nullary then the reduced input type would be `{}`, which
+// unfortunately disables excess property checking, hence the conditional. We
+// wrap the condition in tuples to avoid distributing over the union.
+//
+// It should be noted that excess property checking won't help us if the input
+// object is first defined elsewhere and then provided to our function. This
+// should be considered an unsafe, unsupported use of this API.
```

Comment:
Wasn't this already a limitation?

Reply:
Correct, it just seemed worth writing down.